src/hotspot/share/gc/parallel/psCompactionManager.cpp

```diff
@@ -56,7 +59,8 @@ PreservedMarksSet* ParCompactionManager::_preserved_marks_set = nullptr;
 
 ParCompactionManager::ParCompactionManager(PreservedMarks* preserved_marks,
                                            ReferenceProcessor* ref_processor)
-  : _mark_and_push_closure(this, ref_processor) {
+  :_partial_array_splitter(_partial_array_state_manager, ParallelGCThreads),
```

Comment:
I feel it's best to avoid using `ParallelGCThreads` directly; instead, pass in `parallel_gc_threads` in `new ParCompactionManager` below. WDYT?